### ISSUE TITLE
[BUG]: Duplicate labels creation is possible

### ISSUE THREAD

**Issue by Mitecon** [martinrotter/rssguard#972]
### Brief description of the issue

I've found that when I sometimes accidentally go to create a label I thought I'd missed that it's possible to create a label with an exact duplicate name as one that already exists.

### How to reproduce the bug?

Create a label, then create another label with the exact same name.

### What was the expected result?

Maybe some text along the lines of 'This label already exists', perhaps along with a red X when you go to click 'OK'.

### What actually happened?

RSS Guard accepts any duplicates without question.

I deleted the duplicate and it was fine. It took a fair few seconds, though. It didn't seem to cause any problems, presumably because it would exist in the database with a different ID.

However, what would happen if RSS Guard was running a filter that sent articles to a label where there were two labels with exactly the same name? Would it just apply the ID of the label to each article? Or would there be some internal race to figure out which to use? Maybe there could be some sort of unforeseen conflict that isn't known about yet.

I propose a feature request to check for existing label names and prevent creation of duplicates.

### Debug log

Not really relevant.

### Operating system and version

* OS: Manjaro
* RSS Guard version: I imagine every version does this

**Comment by martinrotter** [comment]
Yes, at this point RSS Guard DB does indeed allow creating labels with duplicate titles. I do not really see any reason why banning that.

If you create more labels with same name and then search label in article filters, RSS Guard would very likely return the older label which was created first.

**Comment by martinrotter** [comment]
You are right. I will add "UNIQUE" SQL condition on label title, but will not re-define existing data, therefore the fix will apply for all new installations of RSS Guard which seems reasonable for this kind of issue., Allright?

**Comment by martinrotter** [comment]
fixed
